### PR TITLE
Cosmetic clean-up of ZM

### PR DESCRIPTION
**Pull request by whannah1**
The Zhang-McFarlane code has suffered a lack of consistency and organization for decades. The current version has many outdated comments and sections that are no longer used. Various authors have modified it over the years and often imposed their own conventions, which has been the source of much confusion and frustration for those wanting to study or modify the code. This PR provides a major cosmetic clean-up of the ZM code, but it is not exhaustive. Some of the quirks of the code have been left intact, such as the lack variable descriptions. Hopefully the enhanced readability will facilitate further refinement. 

[BFB]

### CODE REVIEW

**Comment by whannah1** [comment]
atm_developer passes on chrysalis

File: components/eam/src/physics/cam/zm_conv_intr.F90 [hunk 1]
```diff
@@ -54,249 +49,210 @@ module zm_conv_intr
...
-
+   ! DCAPE-UPL
    if (trigdcape_ull .or. trig_dcape_only) then
-    ! temperature from physics in n-1 time step
-    call pbuf_add_field('T_STAR','global',dtype_r8,(/pcols,pver/), t_star_idx)
-    ! moisturetendency from physics in n-1 time step 
-    call pbuf_add_field('Q_STAR','global',dtype_r8,(/pcols,pver/), q_star_idx)
+      ! temperature from physics in n-1 time step
+      call pbuf_add_field('T_STAR','global',dtype_r8,(/pcols,pver/), t_star_idx)
+      ! moisturetendency from physics in n-1 time step
```
Comment (wlin7) [inline components/eam/src/physics/cam/zm_conv_intr.F90:82]: Preexisting inaccurate comment.  it is certainly not tendency.

Comment (whannah1) [reply components/eam/src/physics/cam/zm_conv_intr.F90:82]: ah right, I'll fix this.

**Review by rljacob — APPROVED** [review]
Code cleanup is good.

File: components/eam/src/physics/cam/zm_conv.F90 [hunk 2]
```diff
@@ -1720,417 +1456,97 @@ subroutine momtran(lchnk, ncol, &
...
-    ! Output apparent force on the mean flow from pressure gradient
+               ! Output apparent force on the mean flow from pressure gradient
                pguall(ii,k,m) = -pgu(i,k)
                pgdall(ii,k,m) = -pgd(i,k)
                icwu(ii,k,m)   =  conu(i,k)
                icwd(ii,k,m)   =  cond(i,k)
             end do
          end do
 
-          ! Calculate momentum flux in units of mb*m/s2
```
Comment (wlin7) [inline components/eam/src/physics/cam/zm_conv.F90:1757]: Note that the reformatted codes here on to the end of the subroutine momtran are all good, though they appear scattering between various segments of the unwanted (thus deleted) subroutine 'buoyan'.

File: components/eam/src/physics/cam/zm_conv.F90 [hunk 3]
```diff
@@ -2480,7 +1845,9 @@ subroutine cldprp(lchnk   , &
    end do
    do i = 1,il2g
       if (j0(i) < jb(i)) then
-         if (f(i,j0(i)) < 1.E-6_r8 .and. f(i,j0(i)+1) > f(i,j0(i))) j0(i) = j0(i) + 1
+         if (f(i,j0(i)) < 1.E-6_r8 .and. f(i,j0(i)+1) > f(i,j0(i))) then
```
Comment (wlin7) [inline components/eam/src/physics/cam/zm_conv.F90:1848]: Just to note that this change goes the opposite way of the changes elsewhere that contain only a single statement within the if-endif block. Not really requesting to revert it.

Comment (whannah1) [reply components/eam/src/physics/cam/zm_conv.F90:1848]: yea, I might have been a little inconsistent in this respect. In this particular case I was thinking the block is more readable when broken into 2 lines. I can see the argument that it's better to use single lines wherever possible though.

**Review by wlin7 — APPROVED** [review]
With visual inspection:  the changes are to clean up comments, tidy up the alignment of the active codes. The codes that are deleted ( a few standalone lines or whole obsolete subroutines) are verified to be not in use.

While all the changes look good for the current code base, suggest to hold on merging for a while to see if any potential complication for upcoming merge of v3 atm features that also touch these two files extensively.

**Pull request closed without merging**